django_gamification/models.py

```diff
@@ -109,6 +109,27 @@ def save(self, *args, **kwargs):
                 badge.points = self.points
                 badge.save()
 
+    def award(self, interface):
+        """
+        Awards this badge to a user.
+        """
+        badges = interface.badge_set.filter(badge_definition=self)
+        badge = None
+        # There should only be one. Otherwise something horrible has happened.
+        # We can handle not having one, where we just need to create one.
+        if len(badges) == 1:
```

Comment:
Once the `unique_together` constraint mentioned on line 128 is implemented, this block can be simplified like so:

```python
if badges.exists():
    badge = badge.first()
    if badge.acquired:
        return
else:
    badge = Badge.objects.create_badge(self, interface)
```